src/Sarif/Errors.cs

```diff
@@ -74,7 +74,7 @@ public static void LogExceptionLoadingPdb(IAnalysisContext context, Exception ex
                     exception,
                     false,
                     context.TargetUri.GetFileName(),
-                    context.Rule.Name));
+                    context.Rule.Name.Text));
```

Comment:
>Text [](start = 38, length = 4)

Existing bug, introduced when we introduced the message object. I noticed it when an unrelated bug in one of the rules I was restoring caused one of the messages in this file to be generated.

Reply:
this is a tough one. if we leave CreateNotification as weakly typed, everyone will continue to fall into this trap. The introduction of the Message object makes it super easy to recompile with an error. If we make our helpers more strongly typed, the call sites are more verbose, everyone needs to explicitly call ToString(). Your thoughts on how to address?




---
In reply to: [210439893](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210439893) [](ancestors = 210439893)

Reply:
Check it out, this bug shipped in current BinSkim. Bad news.

              "message": {
                "text": "'MixedMode_x86_VS2013_MissingPdb.dll' was not evaluated for check 'Microsoft.CodeAnalysis.Sarif.Message' because its PDB could not be loaded. (E_PDB_NOT_FOUND (File not found))"
              },

---
In reply to: [210450798](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210450798) [](ancestors = 210450798,210439893)

Reply:
Ugh.

---
In reply to: [210451144](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210451144) [](ancestors = 210451144,210450798,210439893)